contracts/src/upgrades/IPTokenStaking.sol

```diff
@@ -0,0 +1,547 @@
+        // Use unsetOperator to remove an operator, not setting to zero address
+        require(operator != address(0), "IPTokenStaking: zero input address");
+        emit SetOperator(msg.sender, operator);
+    }
+
+    /// @notice Removes current operator for a delegator.
+    function unsetOperator() external payable chargesFee {
+        emit UnsetOperator(msg.sender);
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+    //                     Staking Configuration functions                    //
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice Set/Update the withdrawal address that receives the withdrawals.
+    /// @param newWithdrawalAddress EVM address to receive the  withdrawals.
+    function setWithdrawalAddress(address newWithdrawalAddress) external payable chargesFee {
+        require(newWithdrawalAddress != address(0), "IPTokenStaking: zero input address");
+        emit SetWithdrawalAddress({
+            delegator: msg.sender,
+            executionAddress: bytes32(uint256(uint160(newWithdrawalAddress))) // left-padded bytes32 of the address
+        });
+    }
+
+    /// @notice Set/Update the withdrawal address that receives the stake and reward withdrawals.
+    /// @dev To prevent spam, only delegators with stake can call this function with cool-down time.
+    /// @param newRewardsAddress EVM address to receive the stake and reward withdrawals.
+    function setRewardsAddress(address newRewardsAddress) external payable chargesFee {
+        require(newRewardsAddress != address(0), "IPTokenStaking: zero input address");
+        emit SetRewardAddress({
+            delegator: msg.sender,
+            executionAddress: bytes32(uint256(uint160(newRewardsAddress))) // left-padded bytes32 of the address
+        });
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+    //                          Validator Creation                            //
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice Entry point for creating a new validator with self delegation.
+    /// @dev The caller must provide the compressed public key that matches the expected EVM address.
+    /// Use this method to make sure the caller is the owner of the validator.
+    /// @param validatorCmpPubkey 33 bytes compressed secp256k1 public key of validator.
+    /// @param moniker The moniker of the validator.
+    /// @param commissionRate The commission rate of the validator.
+    /// @param maxCommissionRate The maximum commission rate of the validator.
+    /// @param maxCommissionChangeRate The maximum commission change rate of the validator.
+    /// @param supportsUnlocked Whether the validator supports unlocked staking.
+    /// @param data Additional data for the validator.
+    function createValidator(
```

Comment:
We don't need to add the `whenNotPaused` modifier here?

Reply:
I see your point @0xHansLee .
Our initial thought was to start with whatever stops a possible loss of funds from the users.
Revisiting this I think we could go with 2 options
- Every non admin methos is pausable but `unstake`
- Every non admin method is pausable

Always allow users to withdraw during an attack vs the attack may be exploiting the unstake logic so nothing should come out.

I think we can start with everything pausable and post in the Forum, see how community feels